bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java

```diff
@@ -39,12 +39,19 @@
   public static final String BIGTABLE_DATA_HOST_DEFAULT = "bigtable.googleapis.com";
   public static final int DEFAULT_BIGTABLE_PORT = 443;
 
-  public static final int BIGTABLE_DATA_CHANNEL_COUNT_DEFAULT = 6;
+  public static final int BIGTABLE_DATA_CHANNEL_COUNT_DEFAULT = getDefaultDataChannelCount();
   public static final int BIGTABLE_CHANNEL_TIMEOUT_MS_DEFAULT =
       (int) TimeUnit.MILLISECONDS.convert(30, TimeUnit.MINUTES);
 
   private static final Logger LOG = new Logger(BigtableOptions.class);
 
+  private static int getDefaultDataChannelCount() {
+    // 10 Channels seemed to work well on a 4 CPU machine, and this seems to scale well for higher
+    // CPU machines
+    int availableProcessors = Runtime.getRuntime().availableProcessors();
+    return (int) Math.max(1, Math.ceil(availableProcessors * 2.5d));
```

Comment:
availableProcessors() isn't supposed to return 0, but I guess you don't trust that, and you are checking that we have atleast 1 channel -  in the same light do you also want some upper bound check as well? Say no more than 250 channels ?